File: examples/symmetry/src/proxy/client/proxy_client.c
#include <stddef.h>

typedef struct { const char *proxy; unsigned long request_count; } proxy_client_t;
void proxy_client_connect(proxy_client_t *c, const char *proxy) { c->proxy = proxy; c->request_count = 0; }
const char *proxy_client_request(proxy_client_t *c, const char *target, const char *payload) { (void)target; c->request_count++; return payload; }

File: examples/symmetry/src/proxy/client/ProxyClient.java
package symmetry.proxy.client;

public final class ProxyClient {
    private String proxy;
    private long requestCount;
    public void connect(String proxy) { this.proxy = proxy; }
    public String request(String target, String payload) { requestCount++; return payload; }
    public String getProxy() { return proxy; }
    public long getRequestCount() { return requestCount; }
}
